### PR TITLE
Add sort options to playback requests

### PR DESCRIPTION
**Pull request by v0idMrK**
**Changes**
Added getting the SortOptions on shortcuts.js and list.js files and sending them in the playbackmanager.js play method

**Issues**
Fixes https://github.com/jellyfin/jellyfin-web/issues/4841

### CODE REVIEW

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -132,3 +132,4 @@
  - [jomp16](https://github.com/jomp16)
  - [Leon de Klerk](https://github.com/leondeklerk)
  - [CrispyBaguette](https://github.com/CrispyBaguette)
+ - [v0idMrK](https://github.com/v0idMrK)
```
Comment (dmitrylyzo) [inline CONTRIBUTORS.md:135]: Should be in another list (Jellyfin contributors).

File: src/components/shortcuts.js [hunk 2]
```diff
@@ -200,12 +213,12 @@ function executeAction(card, target, action) {
         });
     } else if (action === 'play' || action === 'resume') {
         const startPositionTicks = parseInt(card.getAttribute('data-positionticks') || '0', 10);
-
```
Comment (dmitrylyzo) [inline src/components/shortcuts.js:203]: I would keep this empty line, so the code doesn't look too tight. And less unnecessary changes.

File: src/components/playback/playbackmanager.js [hunk 3]
```diff
@@ -1806,12 +1806,18 @@ class PlaybackManager {
                     MediaTypes: 'Audio'
                 });
             } else if (firstItem.MediaType === 'Photo') {
+                const sortOptions = options.sortOptions || {};
+                let sortByValue = options.shuffle ? 'Random' : sortOptions.sortBy;
+                if (sortByValue == null) {
+                    sortByValue = 'SortName';
+                }
```
Comment (dmitrylyzo) [inline src/components/playback/playbackmanager.js:1813]: Looking at other sections, can't we use this on each section (with few adjustments)? Does it work?

File: CONTRIBUTORS.md (hunk 1 above)
Comment (v0idMrK) [reply CONTRIBUTORS.md:135]: Oh maybe I misread on the contribution guide.

Will check later today thank you :)

File: src/components/shortcuts.js (hunk 2 above)
Comment (v0idMrK) [reply src/components/shortcuts.js:203]: I think lint got mad at me earlier by keeping this as a newline?

Maybe it was because there was a few tabs in them? Will also test later today!

Thank you :)

File: src/components/playback/playbackmanager.js (hunk 3 above)
Comment (v0idMrK) [reply src/components/playback/playbackmanager.js:1813]: I haven't tried because I do not use it for music, etc... But I can try and check later today also.


Thank you for the feedback!

**Comment by v0idMrK** [comment]
@dmitrylyzo Just updated this with improvements suggested by your comments. 

Also added the same technique to the other media types on playbackmanager.js. Tested them and they seemed to work fine.

Please let me know if you have any other sugestion! Thank you

File: src/components/shortcuts.js [hunk 4]
```diff
@@ -175,6 +184,10 @@ function executeAction(card, target, action) {
         id = card.getAttribute('data-id');
     }
 
+    const itemsContainer = dom.parentWithClass(card, 'itemsContainer');
+
+    const parentId = itemsContainer.getAttribute('data-parentid');
+
     const item = getItemInfoFromCard(card);
```
Comment (dmitrylyzo) [inline src/components/shortcuts.js:191]: When playing a photo album from the library, the sort from the library is used. Since playing a photo album doesn't (shouldn't) queue its neighbors, we should use its sort rather than the parent sort.

At the same time, to change the sort order of an album, we have to go into it, which is not very convenient. :man_shrugging: 
```suggestion
    const item = getItemInfoFromCard(card);

    const itemsContainer = dom.parentWithClass(card, 'itemsContainer');

    const parentId = item.IsFolder ? item.Id : itemsContainer.getAttribute('data-parentid');
```

File: src/components/shortcuts.js [hunk 5]
```diff
@@ -165,6 +166,14 @@ function showPlayMenu(card, target) {
     });
 }
 
+function getSortValues(parentId) {
+    const basekey = 'items-' + parentId + '-Folder';
+    return {
+        sortBy: userSettings.getFilter(basekey + '-sortby'),
+        sortOrder: userSettings.getFilter(basekey + '-sortorder') === 'Descending' ? 'Descending' : 'Ascending'
+    };
+}
```
Comment (dmitrylyzo) [inline src/components/shortcuts.js:175]: This is probably a candidate for extraction into some utils module (`utils/`) - it is also used by the list.

**Review by dmitrylyzo — COMMENTED** [review]
We should add sort here (`Play` in the context menu):
https://github.com/jellyfin/jellyfin-web/blob/1e47abb88f3ee56b4e7285d7c75c596a5bd1d40f/src/components/itemContextMenu.js#L592-L595

File: src/components/shortcuts.js (hunk 4 above)
Comment (v0idMrK) [reply src/components/shortcuts.js:191]: Cool!

Looks good to me. Later today I will look into this and test it out.

Thank you

File: src/components/shortcuts.js (hunk 5 above)
Comment (v0idMrK) [reply src/components/shortcuts.js:175]: I agree. I'm new to the code so I had no idea about the existence of utils modules.

Will port the code there.

Thanks!

Comment (dmitrylyzo) [reply src/components/shortcuts.js:175]: I think we also can move this to `userSettings` as `getSortValues(basekey)` and `setSortValues(basekey)` - just to limit the use of those magic suffixes.

**Comment by v0idMrK** [comment]
@dmitrylyzo Hey, applied the changes. Let me know if anything else is missing :D

Thank you for your feedback.

File: src/controllers/list.js [hunk 6]
```diff
@@ -962,10 +962,7 @@ class ItemsView {
 
     getSortValues() {
         const basekey = this.getSettingsKey();
-        return {
-            sortBy: userSettings.getFilter(basekey + '-sortby') || this.getDefaultSortBy(),
```
Comment (dmitrylyzo) [inline src/controllers/list.js:964]: We need `this.getDefaultSortBy()`. Maybe pass it as `defaultSortBy` argument?

**Comment by dmitrylyzo** [comment]
> Let me know if anything else is missing

We should add sort for `Play` in the context menu: https://github.com/jellyfin/jellyfin-web/pull/4786#pullrequestreview-1633305594
At least I didn't notice any more.

There is also adding to the play queue, but it doesn't work for photos, so we can leave it for the future.

File: src/controllers/list.js (hunk 6 above)
Comment (v0idMrK) [reply src/controllers/list.js:964]: Going to implement your suggestions. Thank you once again :pray:

**Comment by grafixeyehero** [comment]
@dmitrylyzo @v0idMrK 
A quick Question I wonder what is the use of  `queryOptions`?

**Comment by v0idMrK** [comment]
> @dmitrylyzo @v0idMrK A quick Question I wonder what is the use of `queryOptions`?
> 
> https://github.com/jellyfin/jellyfin-web/blob/055a5008513dc2f1cc370bfbdc5b67c3af1ea8f4/src/components/playback/playbackmanager.js#L1789
> 
> https://github.com/jellyfin/jellyfin-web/blob/055a5008513dc2f1cc370bfbdc5b67c3af1ea8f4/src/components/playback/playbackmanager.js#L1852-L1860

I will try to check today. If I don't have time, then I will go after it tomorrow.

Thank you for the feedback!

File: src/components/shortcuts.js [hunk 7]
```diff
@@ -177,6 +178,10 @@ function executeAction(card, target, action) {
 
     const item = getItemInfoFromCard(card);
 
+    const itemsContainer = dom.parentWithClass(card, 'itemsContainer');
+
+    const sortParentId = 'items-' + (item.IsFolder ? item.Id : itemsContainer.getAttribute('data-parentid')) + '-Folder';
```
Comment (dmitrylyzo) [inline src/components/shortcuts.js:183]: `itemsContainer` can be null - LiveTV Guide.
```suggestion
    const sortParentId = 'items-' + (item.IsFolder ? item.Id : itemsContainer?.getAttribute('data-parentid')) + '-Folder';
```

**Review by dmitrylyzo — COMMENTED** [review]
> A quick Question I wonder what is the use of `queryOptions`?

It doesn't seem to be passed anywhere in `options`, but good catch.
In theory it could be used to adjust the base query (only folders are currently using it).

Looking at `list.js`, the sort options (`SortBy` and `SortOrder`) can be placed in `queryOptions`.
For reference:
https://github.com/jellyfin/jellyfin-web/blob/055a5008513dc2f1cc370bfbdc5b67c3af1ea8f4/src/controllers/list.js#L77-L78

In the `list.js` and `shortcut.js`, we can put sort options as `queryOptions.SortBy` and `queryOptions.SortOrder`.
In the `playbackmanager.js`, we can drop `getSortOptions` and revert my suggestion (https://github.com/jellyfin/jellyfin-web/pull/4786#discussion_r1328672537, most of https://github.com/jellyfin/jellyfin-web/pull/4786/commits/f922742d862e0111e785efd0fb8002098be8de0a), and instead merge `queryOptions` using `mergePlaybackQueries`.

I made test changes: 
[test-photo.diff.txt](https://github.com/jellyfin/jellyfin-web/files/12672204/test-photo.diff.txt)

Btw, SyncPlay duplicates the playback manager. So it should be revised as well.
https://github.com/jellyfin/jellyfin-web/blob/055a5008513dc2f1cc370bfbdc5b67c3af1ea8f4/src/plugins/syncPlay/core/Helper.js#L107
But it doesn't seem to support Photos, so I am not sure if we should fix it right now.

**Comment by grafixeyehero** [comment]
@dmitrylyzo thanks for the clarification. the test change is better in my opinion

**Comment by v0idMrK** [comment]
@dmitrylyzo @grafixeyehero Hi!

Implemented the suggestions. Anything else please let me know!

Thank you for the feedback :+1:

File: src/components/itemContextMenu.js [hunk 8]
```diff
@@ -589,9 +614,12 @@ function play(item, resume, queue, queueNext) {
             serverId: item.ServerId
         });
     } else {
+        const sortParentId = item.IsFolder ? ('items-' + item.Id) : libraryMenu.getTopParentId() + '-' + getSettingsKey(item);
+        const sortValues = userSettings.getSortValues(sortParentId);
         playbackManager[method]({
             items: [item],
-            startPositionTicks: startPosition
+            startPositionTicks: startPosition,
+            queryOptions: sortValues
```
Comment (grafixeyehero) [inline src/components/itemContextMenu.js:622]:
```suggestion
            queryOptions: {
                 SortBy: sortValues.sortBy, 
                 SortOrder: sortValues.sortOrder
              }
```

File: src/components/shortcuts.js [hunk 9]
```diff
@@ -205,7 +210,8 @@ function executeAction(card, target, action) {
             playbackManager.play({
                 ids: [playableItemId],
                 startPositionTicks: startPositionTicks,
-                serverId: serverId
+                serverId: serverId,
+                queryOptions: userSettings.getSortValuesLegacy(sortParentId, 'SortName')
```
Comment (grafixeyehero) [inline src/components/shortcuts.js:214]: here too

File: src/controllers/list.js [hunk 10]
```diff
@@ -722,11 +722,13 @@ class ItemsView {
...
         function play() {
             const currentItem = self.currentItem;
+            const values = self.getSortValues();
 
             if (currentItem && !self.hasFilters) {
                 playbackManager.play({
                     items: [currentItem],
-                    autoplay: true
+                    autoplay: true,
+                    queryOptions: values
```
Comment (grafixeyehero) [inline src/controllers/list.js:731]: here too

**Review by grafixeyehero — CHANGES_REQUESTED** [review]

File: src/components/itemContextMenu.js (hunk 8 above)
Comment (dmitrylyzo) [reply src/components/itemContextMenu.js:622]: It could be a `PhotoAlbum`. The idea is to get its sort options (if any), as if you went into it first and then click `play all`.

File: src/controllers/list.js (hunk 10 above)
Comment (dmitrylyzo) [reply src/controllers/list.js:731]: here `s/sortValues/values/`

File: src/components/playback/playbackmanager.js [hunk 11]
```diff
@@ -1839,15 +1842,15 @@ class PlaybackManager {
...
             } else if (firstItem.Type === 'MusicGenre') {
-                promise = getItemsForPlayback(serverId, {
+                promise = getItemsForPlayback(serverId, mergePlaybackQueries({
                     GenreIds: firstItem.Id,
                     Filters: 'IsNotFolder',
                     Recursive: true,
                     SortBy: options.shuffle ? 'Random' : 'SortName',
                     MediaTypes: 'Audio'
-                });
+                }));
```
Comment (dmitrylyzo) [inline src/components/playback/playbackmanager.js:1853]:
```suggestion
                }, queryOptions));
```

File: src/scripts/settings/userSettings.js [hunk 12]
```diff
@@ -622,6 +622,31 @@ export class UserSettings {
     getFilter(key) {
         return this.get(key, true);
     }
+
+    /**
+     * Gets the current sort values (Legacy - Non-JSON)
+     * (old views such as list.js [Photos] will
+     * use this one)
+     * @param {string} key - Filter key.
```
Comment (dmitrylyzo) [inline src/scripts/settings/userSettings.js:630]:
```suggestion
     * @param {string} key - Filter key.
     * @param {string} defaultSortBy - Default SortBy value.
```

File: src/components/itemContextMenu.js (hunk 8 above)
Comment (grafixeyehero) [reply src/components/itemContextMenu.js:622]: play all executed in shortcuts.js at line 125

Comment (v0idMrK) [reply src/components/itemContextMenu.js]: Removed.

Thank you for the help :+1:

File: src/components/playback/playbackmanager.js (hunk 11 above)
Comment (v0idMrK) [reply src/components/playback/playbackmanager.js:1853]: Removed.

Thank you for the help :+1:

File: src/controllers/list.js (hunk 10 above)
Comment (v0idMrK) [reply src/controllers/list.js]: Removed.

Thank you for the help :+1:

File: src/components/shortcuts.js (hunk 9 above)
Comment (v0idMrK) [reply src/components/shortcuts.js:214]: Are you sure about this one? I tried removing it and the photos sorting stops working :-/

Thank you for the feedback :)

File: src/components/playback/playbackmanager.js (hunk 11 above)
Comment (v0idMrK) [reply src/components/playback/playbackmanager.js]: Done :-)

Thank you!

File: src/scripts/settings/userSettings.js (hunk 12 above)
Comment (v0idMrK) [reply src/scripts/settings/userSettings.js:630]: Done :-)

Thank you!

File: src/controllers/list.js (hunk 10 above)
Comment (dmitrylyzo) [reply src/controllers/list.js]: > Removed.

This is `Play All`, right? Here we need to pass sort options so that we play the current folder in the correct order.

Comment (v0idMrK) [reply src/controllers/list.js]: Yep, you are right. It cannot be removed. I forgot to test that one.

Will be restoring it.

Thank you :-)

File: src/components/itemContextMenu.js (hunk 8 above)
Comment (dmitrylyzo) [reply src/components/itemContextMenu.js]: To test what I was talking about:
1. Go to some photo album.
2. Change its sort order - needs to be noticeable.
3. Go one level up.
4. Click `...` button over that photo album.
5. Click `Play` from the menu.
6. It plays in default order (`SortName Ascending`). _After reverting changes._

The question is whether we want it to play in the default order or in a customized order.

Comment (v0idMrK) [reply src/components/itemContextMenu.js]: Well. It does save in the localsettings it's last sort order for that Album so I don't know. 

Either way it doesn't seem to be like it's bad behavior. One could argue that if you wanted to play without the default order you should go to the photo album itself?

Let me know what yall think.

File: src/controllers/list.js (hunk 10 above)
Comment (v0idMrK) [reply src/controllers/list.js]: Restored :-)

Thanks!

**Review by grafixeyehero — CHANGES_REQUESTED** [review]

File: src/components/itemContextMenu.js [hunk 13]
```diff
@@ -9,6 +9,8 @@ import itemHelper from './itemHelper';
 import { playbackManager } from './playback/playbackmanager';
 import ServerConnections from './ServerConnections';
 import toast from './toast/toast';
+import * as userSettings from '../scripts/settings/userSettings';
+import libraryMenu from '../scripts/libraryMenu';
```
[the remark this answers is not in the document]
Comment (v0idMrK) [reply src/components/itemContextMenu.js]: Done :-)

Thank you!

**Review by grafixeyehero — APPROVED** [review]

**Comment by grafixeyehero** [comment]
@dmitrylyzo it needs your Approval to Thanks👍

**Review by dmitrylyzo — COMMENTED** [review]
With some changes (see below) it works.

But after reverting `itemContextMenu.js` we have some inconsistency: `Play` from the menu doesn't use sort options. See video (00:25).

https://github.com/jellyfin/jellyfin-web/assets/56478732/fb638f56-f890-4aca-9cad-4a8d4c4e2118

https://github.com/jellyfin/jellyfin-web/pull/4786#discussion_r1332228357

First let's discuss how it should play an item (folder) when started from its card (outside): `Play` button on the card and `Play` menu.
1. It should use its sort options outside like `Play All` inside.
[test-photo2.diff.txt](https://github.com/jellyfin/jellyfin-web/files/12683526/test-photo2.diff.txt)
I left only `Folder` (as in the `shortcuts.js`).
Cons: If we want the default order, we have to go into the folder and change the sort options.
2. It should only use its sort options inside (for non-folder items), i.e. outside - default, inside - custom.
We should apply sort options (its parent sort options) in the `shortcuts.js` only if `!item.IsFolder`.
Cons: If we want a custom order (which we set some time earlier), we have to go into the folder and start playing from the inside.

**Comment by grafixeyehero** [comment]
> With some changes (see below) it works.
> 
> But after reverting `itemContextMenu.js` we have some inconsistency: `Play` from the menu doesn't use sort options. See video (00:25).
> 
>  menu-play-1.mp4 
> [#4786 (comment)](https://github.com/jellyfin/jellyfin-web/pull/4786#discussion_r1332228357)
> 
> First let's discuss how it should play an item (folder) when started from its card (outside): `Play` button on the card and `Play` menu.
> 
> 1. It should use its sort options outside like `Play All` inside.
>    [test-photo2.diff.txt](https://github.com/jellyfin/jellyfin-web/files/12683526/test-photo2.diff.txt)
>    I left only `Folder` (as in the `shortcuts.js`).
>    Cons: If we want the default order, we have to go into the folder and change the sort options.
> 2. It should only use its sort options inside (for non-folder items), i.e. outside - default, inside - custom.
>    We should apply sort options (its parent sort options) in the `shortcuts.js` only if `!item.IsFolder`.
>    Cons: If we want a custom order (which we set some time earlier), we have to go into the folder and start playing from the inside.

I agreed with you. 🙌👍

**Comment by v0idMrK** [comment]
I agree with the above points.

As soon as I have time I will implement them :)

(Not sure if today or tomorrow)

Thank you all for the feedback!

**Comment by v0idMrK** [comment]
@dmitrylyzo Hey! Just pushed the latest changes :+1: 

Please let me know if anything else is missing!

Thank you for the feedback.

File: src/components/itemContextMenu.js [hunk 14]
```diff
@@ -589,9 +614,16 @@ function play(item, resume, queue, queueNext) {
             serverId: item.ServerId
         });
     } else {
+        const sortParentId = item.IsFolder ? ('items-' + item.Id) : libraryMenu.getTopParentId() + '-' + getSettingsKey(item);
+        const sortValues = userSettings.getSortValues(sortParentId);
```
Comment (dmitrylyzo) [inline src/components/itemContextMenu.js:618]: For now, I think we can leave just `Folder` (as in `shortcuts.js`) so they work the same way. And `Folder` requires `getSortValuesLegacy` anyway.

Also, `getTopParentId` is only valid for libraries and is empty for lists. So we should probably use `item.ParentId` (as again in `shortcuts.js`), and if it is a folder, we will get its sort options.
```suggestion
        const sortParentId = 'items-' + (item.IsFolder ? item.Id : item.ParentId) + '-Folder';
        const sortValues = userSettings.getSortValuesLegacy(sortParentId);
```

File: src/components/itemContextMenu.js (hunk 13 above)
Comment (dmitrylyzo) [inline src/components/itemContextMenu.js:13]:
```suggestion
```

File: src/components/itemContextMenu.js [hunk 15]
```diff
@@ -567,6 +569,29 @@ function deleteSeriesTimer(apiClient, item, resolve, command) {
...
+            return 'musicalbums';
+        case 'MusicArtist':
+            return 'musicartists';
+        case 'MusicGenre':
+            return 'genres';
+        case 'MusicPlaylist':
+            return 'musicplaylists';
+    }
+}
+
```
Comment (dmitrylyzo) [inline src/components/itemContextMenu.js:594]:
```suggestion
```

File: src/components/playback/playbackmanager.js [hunk 16]
```diff
@@ -1798,23 +1801,23 @@ class PlaybackManager {
...
             } else if (firstItem.MediaType === 'Photo') {
-                promise = getItemsForPlayback(serverId, {
+                promise = getItemsForPlayback(serverId, mergePlaybackQueries({
                     ParentId: firstItem.ParentId,
                     Filters: 'IsNotFolder',
                     // Setting this to true may cause some incorrect sorting
                     Recursive: false,
-                    SortBy: options.shuffle ? 'Random' : 'SortName',
                     MediaTypes: 'Photo,Video',
+                    SortBy: options.shuffle ? 'Random' : 'SortName',
```
[the remark this answers is not in the document]
Comment (v0idMrK) [reply src/components/playback/playbackmanager.js:1818]: Sorry, did not understand what you meant the first time haha.

Will do all of these suggestions tomorrow!

Thanks :)

**Comment by v0idMrK** [comment]
Sorry for not being able to fix this right away. Been busy lately with stuff in life.

Hopefully next week Im able to finish it :)

**Comment by thornbill** [comment]
no worries @v0idMrK whenever you are able to get to it will be fine!

**Comment by v0idMrK** [comment]
@dmitrylyzo Finally got to it after a while :D

Please let me know if there is anything else missing!

Thank you!

**Review by dmitrylyzo — COMMENTED** [review]
Almost good.

File: src/scripts/settings/userSettings.js [hunk 17]
```diff
@@ -622,6 +622,32 @@ export class UserSettings {
...
+    /**
+     * Gets the current sort values (JSON)
+     * (new views such as MoviesView will use
+     * this one)
+     * @param {string} key - Filter key.
+     * @return {Object} sortOptions object
+     */
+    getSortValues(key) {
+        return this.loadQuerySettings(key, {});
+    }
```
[the remark this answers is not in the document]
Comment (v0idMrK) [reply src/scripts/settings/userSettings.js]: Done @dmitrylyzo  thank you once again for the help!

**Review by dmitrylyzo — APPROVED** [review]
LGTM
Need to update the PR title to suit the changelog.

Found another [regression](https://github.com/jellyfin/jellyfin-web/commit/6341a71fec882726bdffa4ed3460a6cbcf252d6e#diff-988c843b03593a8bc9d8238d39f64082e4275e16c9a5cd8e09816a2cd45398bcR21-R22): `SortMenu` doesn't restore current settings because it expects a PascalCase but receives camelCase.
https://github.com/jellyfin/jellyfin-web/blob/c8bed0564dc7d8998bea8aafafd624b983b52a7b/src/components/sortmenu/sortmenu.js#L21-L22
https://github.com/jellyfin/jellyfin-web/blob/c8bed0564dc7d8998bea8aafafd624b983b52a7b/src/controllers/list.js#L964-L965

This should be fixed in a separate PR. Maybe just use PascalCase for `SortBy` and `SortOrder` everywhere (list, shortcuts, etc.).

File: src/components/playback/playbackmanager.js [hunk 18]
```diff
@@ -145,7 +145,10 @@ function createStreamInfoFromUrlItem(item) {
 }
 
 function mergePlaybackQueries(obj1, obj2) {
-    const query = Object.assign(obj1, obj2);
+    const query = obj1;
+    for (const key in obj2) {
+        if (obj2[key] !== undefined) query[key] = obj2[key];
```
Comment (thornbill) [inline src/components/playback/playbackmanager.js:150]: Maybe just use the lodash merge implementation?

Comment (thornbill) [reply src/components/playback/playbackmanager.js:150]: @v0idMrK what do you think about using lodash for this?

Comment (v0idMrK) [reply src/components/playback/playbackmanager.js:150]: @thornbill 

Hey, sorry for only answering now! Been super busy lately.

It sounds good to me. I will see if today Im able to change it quickly.

Thank you for the heads up!

**Comment by grafixeyehero** [comment]
> LGTM Need to update the PR title to suit the changelog.
> 
> Found another [regression](https://github.com/jellyfin/jellyfin-web/commit/6341a71fec882726bdffa4ed3460a6cbcf252d6e#diff-988c843b03593a8bc9d8238d39f64082e4275e16c9a5cd8e09816a2cd45398bcR21-R22): `SortMenu` doesn't restore current settings because it expects a PascalCase but receives camelCase.
> 
> https://github.com/jellyfin/jellyfin-web/blob/c8bed0564dc7d8998bea8aafafd624b983b52a7b/src/components/sortmenu/sortmenu.js#L21-L22
> 
> 
> https://github.com/jellyfin/jellyfin-web/blob/c8bed0564dc7d8998bea8aafafd624b983b52a7b/src/controllers/list.js#L964-L965
> 
> This should be fixed in a separate PR. Maybe just use PascalCase for `SortBy` and `SortOrder` everywhere (list, shortcuts, etc.).

@dmitrylyzo https://github.com/jellyfin/jellyfin-web/pull/4875

**Comment by v0idMrK** [comment]
@thornbill @dmitrylyzo 

Hey! Sorry still didnt have time!

When I do I will fix it, just wanted you to know that I didnt forget

**Comment by v0idMrK** [comment]
@thornbill @dmitrylyzo

Could you please check it out when you find the time :smiley: 

Thank you !

**Review by thornbill — APPROVED** [review]
